datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -843,6 +846,8 @@ pub struct SortExec {
     common_sort_prefix: Vec<PhysicalSortExpr>,
     /// Cache holding plan properties like equivalences, output partitioning etc.
     cache: PlanProperties,
+    /// Filter matching the state of the sort for dynamic filter pushdown
+    filter: Option<Arc<DynamicFilterPhysicalExpr>>,
```

Comment:
What isn't really clear to me from this PR - will this create a filter here regardless whether it can be pushed down or do we keep this as `None` if it can't? As there might be a small overhead of evaluating this filter.

Reply:
It will be created regardless atm. Possibly could be optimized, e.g. by keeping track of refs but not too sure.

Reply:
It _won't_ be created it there's no `limit` applied is the idea.

Reply:
That might make sense for now 🤔
I think once we do https://github.com/apache/datafusion/pull/15697 probably it's not needed to disable computing it.

Reply:
it would also be nice to add a note about when this is field is set or not